src/lib/compile.ml

```diff
@@ -208,15 +208,33 @@ module Compile = struct
     in
     let* build_job = Current.Job.start_with ~pool:build_pool ~level:Mostly_harmless job in
     Current.Job.log job "Using cache hint %S" cache_hint;
-    Capnp_rpc_lwt.Capability.with_ref build_job @@ fun build_job ->
-    let** _ = Current_ocluster.Connection.run_job ~job build_job in
-    let extract_hashes (v_compile, v_linked) line =
+    let extract_hashes ((v_compile, v_linked), retriable_errors) line =
+      let retry_conditions log_line =
+        let retry_on = [
+          "Temporary failure";
+          "Could not resolve host";
+          "rsync: connection unexpectedly closed"
+        ] in
+        List.fold_left
+          (fun acc str -> acc || Astring.String.is_infix ~affix:str log_line) false retry_on
+      in
       (* some early stopping could be done here *)
```

Comment:
It means that when the `compile` and `linked` values are obtained, we could stop parsing the log file (in `Misc.fold_logs`). But it's not very important, it would just be a bit less work for the CI.